### ISSUE TITLE
Wpf cannot working with NavigationService

### ISSUE THREAD

**Issue by DatDevSc** [SuRGeoNix/Flyleaf#300]
Hi

My work is:
1. Two UserControl: UcCamera(video view) and UcSettings(setting)
2. A ListView as Navigation, when ListViewItem changed. Create new instance of UserControl and navigate to that instance by Frame on MainWindow. All UserControl use the same DataContext

When I run the program, right after video display, video disappear then stuck without any error or crash. 

I have reproduced the issue by fork the latest version of your work and add new project at: Solution/Test/Navigation. 
https://github.com/DatDevSc/Flyleaf

Could you please help me to check this issue. 

Thank you so much!

**Comment by SuRGeoNix** [comment]
@DatDevSc The main issue is that you create the controls each time you change your navigation frame. You should have a singleton instances for the UcCamera control and possible disable the virtualization to avoid issues as well? and finally don't destroy the frame (with RemoveBackEntry). I'm looking into it and I will come back

**Comment by DatDevSc** [comment]
Thank you so much, it work really well.

Sorry but I don't have any knowledge about singletons, could I ask about it?

Is singletons mean that I create instance of UcCamera and UcSettings in VmMainWindow.
```
UcCamera CameraInstance = new();
UcSettings SettingsInstance = new();
```
Then navigate by compare in Navigation changed
```
public async void NaviChanged()
{
    if (_Navi.Type == typeof(UcCamera))
    {
        NavigationManager.NavigationService.Navigate(CameraInstance);
        CameraInstance.DataContext = App.Current.MainWindow?.DataContext;
    }
    else if (_Navi.Type == typeof(UcCamera))
    {
        NavigationManager.NavigationService.Navigate(SettingsInstance);
        SettingsInstance.DataContext = App.Current.MainWindow.DataContext;
    }
    await Task.Delay(500);
}
```

Is it correct for singletons?

**Comment by SuRGeoNix** [comment]
@DatDevSc yes. That's the idea anyways. They usually doing this with 'services' that registers the instances when the app initializes and then you request for the service when you need it. Just make sure you don't remove the frame. I'm not a WPF expert and never use the NavigationService but a quick try when I removed this line https://github.com/DatDevSc/Flyleaf/blob/master/Navigation/MainWindow.xaml.cs#L31 was working fine with singletons.

**Comment by DatDevSc** [comment]
Hi @SuRGeoNix, with remove `this.MainFrame.RemoveBackEntry();` work well. But there is another problem with FindAncestor at this line
https://github.com/SuRGeoNix/Flyleaf/blob/72d6c6e5b481bf54450c20bb16dfc1776044bfb8/FlyleafLib/Controls/WPF/FlyleafSharedOverlay.cs#L127

If I add some control overlay the host, when I change the navigation: System.InvalidOperationException: 'The specified Visual is not an ancestor of this Visual.'
I have committed the issue at https://github.com/DatDevSc/Flyleaf.
Could you please help me to check it?

Thank you so much!
